dff/script/extras/conditions/models/remote_api/hf_api_model.py

```diff
@@ -0,0 +1,134 @@
+"""
+HuggingFace API Model
+---------------------------
+
+This module provides the :py:class:`~HFAPIModel` class that allows you
+to use remotely hosted HuggingFace models via the HuggingFace inference API.
+"""
+import time
+import json
+import asyncio
+from typing import Optional
+from urllib.parse import urljoin
+
+import requests
+
+try:
+    import httpx
+
+    hf_api_available = True
+except ImportError:
+    hf_api_available = False
+
+from ...utils import STATUS_SUCCESS, STATUS_UNAVAILABLE
+from ..base_model import BaseModel
+from .async_mixin import AsyncMixin
+
+
+class AbstractHFAPIModel(BaseModel):
+    """
+    Abstract class for an HF API annotator.
+    """
+
+    def __init__(
+        self,
+        model: str,
+        api_key: str,
+        namespace_key: Optional[str] = None,
+        *,
+        retries: int = 60,
+        headers: Optional[dict] = None,
+    ) -> None:
+        super().__init__(namespace_key=namespace_key)
+        self.api_key = api_key
+        self.model = model
+        self.headers = (
+            headers if headers else {"Authorization": "Bearer " + api_key, "Content-Type": "application/json"}
+        )
+        self.retries = retries
+        self.url = urljoin("https://api-inference.huggingface.co/models/", model)
```

Comment:
Isn't there any possible case for self-hosted models? Maybe the link should be moved to the default arguments?